libraries/chain/apply_context.cpp

```diff
@@ -224,4 +224,15 @@ const contracts::table_id_object& apply_context::find_or_create_table( name scop
    });
 }
 
+void apply_context::get_active_producers(account_name* producers, size_t datalen) {
```

Comment:
Part of the purpose of the separation of `apply_context` and `wasm_interface` is that the `apply_context` should have implementations that look like they belong in our native code and the `wasm_interface` is responsible for the shim between that and the webassembly VM.  

In this case, I think I would prefer this to have the signature `vector<account_name> (void) const` (`void` for clarity).

This would mean that the logic for guarding against short arrays etc would be part of the `wasm_interface` glue that unfortunately has to deal with data buffers instead of proper containers.